oak-search-elastic/src/main/java/org/apache/jackrabbit/oak/plugins/index/elastic/query/ElasticSpellcheckIterator.java

```diff
@@ -102,24 +100,25 @@ private void loadSuggestions() {
                             .source(searchSource))
                     .reduce(new MultiSearchRequest(), MultiSearchRequest::add, (ms, ms2) -> ms);
 
-            ArrayList<FulltextResultRow> results = new ArrayList<>();
-            MultiSearchResponse res = indexNode.getConnection().getClient().msearch(multiSearch, RequestOptions.DEFAULT);
-            for (MultiSearchResponse.Item response : res.getResponses()) {
-                boolean hasResults = false;
-                for (SearchHit doc : response.getResponse().getHits()) {
-                    if (responseHandler.isAccessible(responseHandler.getPath(doc))) {
-                        hasResults = true;
-                        break;
+            if (!multiSearch.requests().isEmpty()) {
+                MultiSearchResponse res = indexNode.getConnection().getClient().msearch(multiSearch, RequestOptions.DEFAULT);
+                ArrayList<FulltextResultRow> results = new ArrayList<>();
+                for (MultiSearchResponse.Item response : res.getResponses()) {
+                    boolean hasResults = false;
+                    for (SearchHit doc : response.getResponse().getHits()) {
+                        if (responseHandler.isAccessible(responseHandler.getPath(doc))) {
+                            hasResults = true;
+                            break;
+                        }
+                    }
+                    String suggestion = suggestionTexts.poll();
```

Comment:
This could be moved inside the condition... actually the flag "hasResults" is not strictly needed and the conditional statement could be moved inside the "if" statement above? Not sure if it works...

                    for (SearchHit doc : response.getResponse().getHits()) {
                        if (responseHandler.isAccessible(responseHandler.getPath(doc))) {
                            String suggestion = suggestionTexts.poll();
                            results.add(new FulltextResultRow(suggestion));
                            break;
                        }
                    }

Reply:
good catch!